iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java

```diff
@@ -37,17 +37,116 @@
+            this.index = index;
+        }
+
+        @Override
+        public Bus getBus(final String id) {
+            final List<Network.BusBreakerView> views = index.getNetworkStream()
+                                                      .map(Network::getBusBreakerView).collect(Collectors.toList());
+            for (final Network.BusBreakerView bb : views) {
+                final Bus bus = bb.getBus(id);
+                if (Objects.nonNull(bus)) {
+                    return index.getBus(bus);
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public Iterable<Bus> getBuses() {
+            return Collections.unmodifiableList(getBusStream().collect(Collectors.toList()));
+        }
+
+        @Override
+        public Stream<Bus> getBusStream() {
+            return index.getNetworkStream()
+                    .map(Network::getBusBreakerView)
+                    .map(Network.BusBreakerView::getBusStream)
+                    .flatMap(stream -> stream)
+                    .map(index::getBus);
+        }
+
+        @Override
+        public Iterable<Switch> getSwitches() {
+            return Collections.unmodifiableList(getSwitchStream().collect(Collectors.toList()));
+        }
+
+        @Override
+        public Stream<Switch> getSwitchStream() {
+            return index.getNetworkStream()
+                    .map(Network::getBusBreakerView)
+                    .map(Network.BusBreakerView::getSwitchStream)
+                    .flatMap(stream -> stream)
+                    .map(index::getSwitch);
+        }
+
+        @Override
+        public int getSwitchCount() {
+            return (int) getSwitchStream().count();
+        }
+    }
+
+    private final BusBreakerViewAdapter busBreakerView;
+
+    private static class BusViewAdapter implements Network.BusView {
+
+        private final MergingViewIndex index;
+
+        BusViewAdapter(final MergingViewIndex index) {
+            this.index = index;
+        }
+
+        @Override
+        public Iterable<Bus> getBuses() {
+            return Collections.unmodifiableList(getBusStream().collect(Collectors.toList()));
+        }
+
+        @Override
+        public Stream<Bus> getBusStream() {
+            return index.getNetworkStream()
+                    .map(Network::getBusView)
+                    .map(Network.BusView::getBusStream)
+                    .flatMap(stream -> stream)
+                    .map(index::getBus);
+        }
+
+        @Override
+        public Bus getBus(final String id) {
+            final List<Network.BusView> views = index.getNetworkStream()
+                    .map(Network::getBusView).collect(Collectors.toList());
+            for (final Network.BusView bv : views) {
+                final Bus bus = bv.getBus(id);
+                if (Objects.nonNull(bus)) {
+                    return index.getBus(bus);
+                }
+            }
+            return null;
```

Comment:
Should be implemented using lambdas:
```
return index.getNetworkStream()
        .map(n -> n.getBusView().getBus(id))
        .findFirst()
        .map(index::getBus)
        .orElse(null);
```
Not totally sure of my proposal, so I let you test it

Reply:
It works ... 
I just add a filter on nullalbe bus